oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthCodeHandler.java

```diff
@@ -123,165 +125,89 @@ public APIGatewayProxyResponseEvent handleRequest(
-                                    clientSession.getClientName()));
-
-            if (Objects.nonNull(session.getVerifiedMfaMethodType())) {
-                dimensions.put("MfaMethod", session.getVerifiedMfaMethodType().getValue());
-            } else {
-                LOG.info(
-                        "No mfa method to set. User is either authenticated or signing in from a low level service");
-            }
-
-            var internalSubjectId = AuditService.UNKNOWN;
+            var subjectId = AuditService.UNKNOWN;
             var rpPairwiseId = AuditService.UNKNOWN;
             String internalCommonPairwiseSubjectId;
             if (docAppJourney) {
                 LOG.info("Session not saved for DocCheckingAppUser");
                 internalCommonPairwiseSubjectId = clientSession.getDocAppSubjectId().getValue();
             } else {
-                var mfaNotRequired =
-                        clientSession
-                                .getEffectiveVectorOfTrust()
-                                .getCredentialTrustLevel()
-                                .equals(CredentialTrustLevel.LOW_LEVEL);
-                var levelOfConfidence = LevelOfConfidence.NONE.getValue();
-                if (clientSession.getEffectiveVectorOfTrust().containsLevelOfConfidence()) {
-                    levelOfConfidence =
-                            clientSession
-                                    .getEffectiveVectorOfTrust()
-                                    .getLevelOfConfidence()
-                                    .getValue();
-                }
-                dimensions.put("MfaRequired", mfaNotRequired ? "No" : "Yes");
-                dimensions.put("RequestedLevelOfConfidence", levelOfConfidence);
+                authCodeResponseService.processVectorOfTrust(clientSession, dimensions);
                 internalCommonPairwiseSubjectId = session.getInternalCommonSubjectIdentifier();
-                var userProfile =
-                        dynamoService
-                                .getUserProfileByEmailMaybe(session.getEmailAddress())
-                                .orElseThrow(
-                                        () ->
-                                                new UserNotFoundException(
-                                                        "Unable to find user with given email address"));
-                var client =
-                        dynamoClientService
-                                .getClient(clientID.getValue())
-                                .orElseThrow(
-                                        () -> new ClientNotFoundException(clientID.getValue()));
-                internalSubjectId =
-                        Objects.isNull(session.getEmailAddress())
-                                ? AuditService.UNKNOWN
-                                : userProfile.getSubjectID();
+                subjectId = authCodeResponseService.getSubjectId(session);
```

Comment:
I think it's a shame that this means that we duplicate the user profile dynamo lookup. Fine for now, but let's follow this up to see if we can improve this